src/nvim/terminal.c

```diff
@@ -313,6 +313,10 @@ void terminal_resize(Terminal *term, uint16_t width, uint16_t height)
     return;
   }
 
+  if (height == 0 || width == 0) {
+    return;
+  }
```

Comment:
> when splitting a window (win_split_ins in window.c), we first set the new width (win_new_width) which updates the terminal via terminal_resize (but at that time the height hasn't been set yet). After that we set the new height (win_new_height) for the one window, then for the other. This function also calls terminal_resize, with more correct values.

I see. Now I understand why exiting when `width` or `height` equals zero fixes the problem--we called `terminal_resize` before actually knowing what size it should be. I think the new window's dimensions (from `win_split() -> win_alloc()`) have been initialized to zero. If that's the case, it would be more optimal to have something like this in the `FOR_ALL_WINDOWS_IN_TAB` block:
```c
if (wp->w_height == 0) {
  return;  // Window still being initialized. win_new_height() will call this again.
}
```
We only need to check `w_height` `win_alloc()` sets `w_width` to `Rows`.

Sorry I'm delaying this so much. I just want to make sure we're fixing it in the right way.